sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1342,16 +1354,23 @@ void cleanup(ContextContainer c) throws Exception {
               BigQueryOptions options = c.getPipelineOptions().as(BigQueryOptions.class);
               String jobUuid = c.getJobId();
 
+              Optional<String> queryTempDataset = Optional.ofNullable(getQueryTempDataset());
```

Comment:
If dataset is provided by the user we should try to validate (before pipeline submission) that it exists. (unless user specified withoutValidation())

Reply:
Got it, let me add that.

Reply:
Commit added for that. See also reply to another of your comments.